sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations.py

```diff
@@ -469,6 +476,63 @@ def _populate_arm_paramaters(self, workspace: Workspace) -> Tuple[dict, dict, di
                 group_name,
             )
         else:
+            # if workspace is located in a region where app insights is not supported, we do this swap
+            if workspace.location in ["westcentralus", "eastus2euap", "centraluseuap"]:
```

Comment:
Is this needed given the ARM template checks for the app insights location too?

Reply:
I do this check to get the location to use for default Resource Group and default Log Analytics needed for app insights as well for the alternate region if necessary. It is needed for getting the deployments for those.